### PR TITLE
Implement throwing of CommandFailedStackException in SequentialCommandBus.

### PR DESCRIPTION
**Pull request by coderbyheart**

### CODE REVIEW

File: src/LiteCQRS/Bus/SequentialCommandBus.php [hunk 1]
```diff
@@ -75,6 +79,12 @@ public function handle($command)
             $this->executing = false;
             $first = false;
         }
+
+        if (!empty($this->commandStack)) return; // Execute other pending commands
+        if (empty($this->exceptionStack)) return;
+        $e = new CommandFailedStackException($this->exceptionStack);
+        $this->exceptionStack = array();
+        throw $e;
```
Comment (henrikbjorn) [inline src/LiteCQRS/Bus/SequentialCommandBus.php:87]: The code block above does not follow the coding style of the rest of the library, which uses PSR-1/2

File: src/LiteCQRS/Bus/SequentialCommandBus.php [hunk 2]
```diff
@@ -82,6 +92,7 @@ protected function handleException($e, $first)
         if ($first) {
             throw $e;
         }
+        $this->exceptionStack[] = $e;
```
Comment (henrikbjorn) [inline src/LiteCQRS/Bus/SequentialCommandBus.php:95]: Missing newline before the new line.

**Comment by beberlei** [comment]
Subcommand exceptions should not be thrown imho. Say you have a command that triggers $N events, and several of those trigger new commands. If one command fails, then this is just the way it is, but everything else should not be stopped because of it. It will create very weird inconsistent states.

**Comment by coderbyheart** [comment]
How can I get notified of an exception in the command chain then?

**Comment by coderbyheart** [comment]
Ok. CQRS introduces eventual consistency. CQRS also means that you should assume that every command succeeds. To verify that a command worked, use a query. Failed commands should notify to user, though.

I've created PR #31 which clarifies the behaviour of SequentialCommandBus.

**Comment by coderbyheart** [comment]
How about we create a CommandFailed event, so exceptions would not be silently dropped?

**Comment by henrikbjorn** [comment]
Just because you use the CommandBus dosent mean you also you the EventBus.

**Pull request closed without merging**